src/script/page/RootProvider.tsx

```diff
@@ -28,8 +28,15 @@ interface RootProviderProps {
   value: MainViewModel;
 }
 
-const RootProvider: FC<RootProviderProps> = ({children, value}) => {
+export const RootProvider: FC<RootProviderProps> = ({children, value}) => {
   return <RootContext.Provider value={value}>{children}</RootContext.Provider>;
 };
 
-export {RootProvider};
+export const useMainViewModel = () => {
+  const context = useContext(RootContext);
+  if (!context) {
+    throw new Error('MainViewModel was not initialised');
+  }
+
+  return context;
+};
```

Comment:
We're always passing `MainViewModel` to `RootProvider` so this should be totally fine and typesafe (no need to check for possible null inside `Conversation` component)